### PR TITLE
[Fix] Samples unavailable on Mac Catalyst

### PR DESCRIPTION
**Pull request by dfeinzimer**
Toolkit AR components are no longer available on Catalyst

### CODE REVIEW

File: Scripts/GenerateSampleViewSourceCode.swift [hunk 1]
```diff
@@ -139,15 +140,26 @@ private let sampleStructs = sampleMetadata
...
-        .filter { $0.category != "Augmented Reality" }
+        [
+            \(ARFilteredEntries)
+        ]
+    #else
+        [
+            \(entries)
+        ]
     #endif
+    }()
```
Comment (dfeinzimer) [inline Scripts/GenerateSampleViewSourceCode.swift:162]: Here's a preview of the new output.

[SamplesApp+Samples.swift.zip](https://github.com/user-attachments/files/17604603/SamplesApp%2BSamples.swift.zip)

It's not great but short of changing the the way `SamplesApp+Samples` is constructed, this is the best short term reformatting I've come up with so far.

**Comment by dfeinzimer** [comment]
The offending PR has been reverted in the Toolkit for now.

**Comment by dfeinzimer** [comment]
@yo1995 I've revised the changes a bit to generate a smaller `SamplesApp+Samples.swift` ([preview](https://github.com/user-attachments/files/17623065/SamplesApp%2BSamples.swift.zip)).

Ready for review.

File: Shared/SamplesApp+Samples.swift.tache [hunk 2]
```diff
@@ -15,8 +15,11 @@
 import SwiftUI
 
 extension SamplesApp {
-    /// The samples for this app.
-    static let samples: [Sample] = [/* samples */]
+    /// The common samples for this app.
+    static let commonSamples: [Sample] = [/* common_samples */]
+
```
Comment (yo1995) [inline Shared/SamplesApp+Samples.swift.tache:20]:
```suggestion
```

File: Shared/Samples/Augment reality to fly over scene/AugmentRealityToFlyOverSceneView.swift [hunk 3]
```diff
@@ -59,6 +60,8 @@ private extension URL {
     }
 }
 
+#if !targetEnvironment(macCatalyst)
 #Preview {
     AugmentRealityToFlyOverSceneView()
 }
+#endif
```
Comment (yo1995) [inline Shared/Samples/Augment reality to fly over scene/AugmentRealityToFlyOverSceneView.swift:67]: I wonder if we should remove the preview - even if the scene view shows, it doesn't convey the concept of this sample when previewed in a simulator 🤔

File: Shared/Samples/Augment reality to collect data/AugmentRealityToCollectDataView.swift [hunk 4]
```diff
@@ -199,6 +202,8 @@ private extension AugmentRealityToCollectDataView {
     }
 }
 
+#if !targetEnvironment(macCatalyst)
 #Preview {
     AugmentRealityToCollectDataView()
 }
+#endif
```
Comment (yo1995) [inline Shared/Samples/Augment reality to collect data/AugmentRealityToCollectDataView.swift:209]: Same comment for this sample - maybe preview can be removed.

File: Scripts/GenerateSampleViewSourceCode.swift [hunk 5]
```diff
@@ -136,25 +137,42 @@ private let sampleStructs = sampleMetadata
...
     .joined(separator: "\n")
 
-private let entries = sampleMetadata
+private let commonSamples = sampleMetadata
+    .filter { $0.category != "Augmented Reality" }
     .map { sample in "\(sample.structName)()" }
     .joined(separator: ",\n        ")
-private let arrayRepresentation = """
+
+private let iOSSamples = sampleMetadata
```
Comment (yo1995) [inline Scripts/GenerateSampleViewSourceCode.swift:145]: Maybe call this `iOSSpecificSamples`, and/or add some comment for both variables on why we separate these samples from the common samples, so it's easier to reason about in the future.

File: Shared/SamplesApp.swift [hunk 6]
```diff
@@ -31,6 +31,11 @@ struct SamplesApp: App {
 // MARK: - License
 
 extension SamplesApp {
+    /// The samples for this app.
+    static var samples: [Sample] {
+        return SamplesApp.commonSamples + SamplesApp.iOSSamples
```
Comment (dfeinzimer) [inline Shared/SamplesApp.swift:36]: Should these be alphabetically sorted?

File: Shared/Samples/Augment reality to fly over scene/AugmentRealityToFlyOverSceneView.swift (hunk 3 above)
Comment (dfeinzimer) [reply Shared/Samples/Augment reality to fly over scene/AugmentRealityToFlyOverSceneView.swift:67]: Seems like the best thing to do is remove AR previews.

File: Shared/SamplesApp.swift (hunk 6 above)
Comment (dfeinzimer) [reply Shared/SamplesApp.swift:36]: Sorry, not sure I follow. At which point in the script do you mean?

Comment (yo1995) [reply Shared/SamplesApp.swift:36]: The `SamplesApp.commonSamples` and `SamplesApp.iOSSamples` can be private variables in the script, and we can expose a computed property `SamplesApp.samples` that is sorted. Does that work?

Comment (yo1995) [reply Shared/SamplesApp.swift:36]: In the template `SamplesApp+Samples.swift` file, have the following

```swift
extension SamplesApp {
    /// The common samples for this app.
    private static let commonSamples: [Sample] = [/* common_samples */]
    /// The iOS samples for this app.
    private static let iOSSamples: [Sample] = [/* ios_samples */]
    /// The samples for this app.
    static let samples = (SamplesApp.commonSamples + SamplesApp.iOSSamples).sorted { $0.name < $1.name }
}
```

And remove the extension in `Shared/SamplesApp.swift`

**Comment by yo1995** [comment]
Hi David, did you branch off the `main` branch? I think this fix should go onto `v.next`. 🤔

**Comment by dfeinzimer** [comment]
Rebased to v.next and resolved merge conflicts, ready for re-review.

**Comment by dfeinzimer** [comment]
Closing in favor of #541 at Ting's request

**Pull request closed without merging**